src/main/java/com/jaspersoft/jasperserver/jaxrs/client/dto/settings/AwsSettings.java

```diff
@@ -0,0 +1,133 @@
+package com.jaspersoft.jasperserver.jaxrs.client.dto.settings;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Created by tetiana.iefimenko on 7/14/2015.
+ */
+public class AwsSettings {
+
+
+    private Boolean productTypeIsEc2;
+    private Boolean isEc2Instance;
+    private Boolean productTypeIsJrsAmi;
+    private List<String> awsRegions;
+    private Boolean productTypeIsMpAmi;
+    private Boolean suppressEc2CredentialsWarnings;
+
+    public AwsSettings() {
+    }
+
+    public AwsSettings(AwsSettings other) {
+        this.productTypeIsEc2 = other.productTypeIsEc2;
+        this.isEc2Instance = other.isEc2Instance;
+        this.productTypeIsJrsAmi = other.productTypeIsJrsAmi;
+        this.awsRegions = new LinkedList<String>(other.awsRegions);
```

Comment:
other.awsRegions can be null and exception will be thrown in this case. Please add check for null.